File: leetcode_hot100/160 相交链表/code.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        curA = headA
        curB = headB
        lenA = 1
        lenB = 1
        gap = 0
        while(curA.next != None):
            lenA += 1
            curA = curA.next
        while(curB.next != None):
            lenB += 1
            curB = curB.next
        if (lenA < lenB):
            curA = headB
            curB = headA
            gap = lenB - lenA
        else:
            curA = headA
            curB = headB
            gap = lenA - lenB
        while(gap != 0):
            gap -= 1
            curA = curA.next
        while(curA != None):
            if (curA == curB):
                return curA
            curA = curA.next
            curB = curB.next
        return None

File: leetcode_hot100/160 相交链表/code.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function getIntersectionNode(headA: ListNode | null, headB: ListNode | null): ListNode | null {
    let curA:ListNode|null = headA;
    let curB:ListNode|null = headB;
    let lenA:number = 1;
    let lenB:number = 1;
    let gap:number = 0;
    while(curA.next !== null){
        lenA++;
        curA=curA.next;
    }
    while(curB.next !== null){
        lenB++;
        curB=curB.next
    }
    if (lenA < lenB){
        gap = lenB-lenA;
        curA = headB;
        curB = headA;
    } else {
        gap = lenA-lenB;
        curA = headA;
        curB = headB;
    }
    while(gap--){
        curA=curA.next;
    }
    while(curA !== null){
        if (curA === curB){
            return curA
        }
        curA=curA.next;
        curB=curB.next;
    }
    return null;
};